framework/tests/src/tests/signatureinference/SignatureInferenceTestAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,162 @@
+    protected class SignatureTestQualifierHierarchy extends
+                MultiGraphQualifierHierarchy {
+
+        public SignatureTestQualifierHierarchy(MultiGraphFactory f) {
+            super(f);
+        }
+
+        @Override
+        public AnnotationMirror leastUpperBound(AnnotationMirror a1,
+                AnnotationMirror a2) {
+            if ((AnnotationUtils.areSameByClass(a1, Sibling1.class)
+                    && AnnotationUtils.areSameByClass(a2, Sibling2.class)) 
+                    || (AnnotationUtils.areSameByClass(a1, Sibling2.class)
+                    && AnnotationUtils.areSameByClass(a2, Sibling1.class))
+                    || (AnnotationUtils.areSameByClass(a1, Sibling1.class)
+                    && AnnotationUtils.areSameByClass(a2, SiblingWithFields.class))
+                    || (AnnotationUtils.areSameByClass(a1, SiblingWithFields.class)
+                    && AnnotationUtils.areSameByClass(a2, Sibling2.class))
+                    || (AnnotationUtils.areSameByClass(a1, Sibling2.class)
+                    && AnnotationUtils.areSameByClass(a2, SiblingWithFields.class))) {
+                return PARENT;
+            }
+            return super.leastUpperBound(a1, a2);
+        }
+        @Override
+        public boolean isSubtype(AnnotationMirror sub, AnnotationMirror sup) {
+            if (AnnotationUtils.areSame(sub, sup)
+                    || AnnotationUtils.areSameByClass(sup, UnknownClass.class)
+                    || AnnotationUtils.areSameByClass(sub, SignatureInferenceBottom.class)
+                    || AnnotationUtils.areSameByClass(sup, Top.class)) {
+                return true;
+            }
+
+            if (AnnotationUtils.areSameByClass(sub, UnknownClass.class)
+                    || AnnotationUtils
+                            .areSameByClass(sup, SignatureInferenceBottom.class)) {
+                return false;
+            }
+
+            if (AnnotationUtils.areSameByClass(sub, Top.class)) {
+                return false;
+            }
+
+            if ((AnnotationUtils.areSameByClass(sub, Sibling1.class)
+                    || AnnotationUtils.areSameByClass(sub, Sibling2.class)
+                    || AnnotationUtils.areSameByClass(sub, SiblingWithFields.class))
+                    && AnnotationUtils.areSameByClass(sup, Parent.class)) {
+                return true;
+            }
+
+            if ((AnnotationUtils.areSameByClass(sub, Sibling1.class)
+                    || AnnotationUtils.areSameByClass(sub, Sibling2.class)
+                    || AnnotationUtils.areSameByClass(sub, SiblingWithFields.class)
+                    || AnnotationUtils.areSameByClass(sub, Parent.class)) 
+                    && AnnotationUtils.areSameByClass(sup, DefaultType.class)){
+                return true;
+            }
+
+            if (AnnotationUtils.areSameByClass(sub, SiblingWithFields.class)
+                    && AnnotationUtils.areSameByClass(sup, SiblingWithFields.class)) {
```

Comment:
I find it clearer to align the && under the "A", that is one character further right than the "(" that encloses it.